src/virustotal.ts

```diff
@@ -32,7 +34,21 @@ export class VirusTotal {
     });
   }
 
-  files(filename: string): Promise<UploadData> {
+  private async getURL(apiKey: string | undefined): Promise<string> {
+    const instance: AxiosInstance = axios.create({
+      baseURL: '',
+      headers: {
+        Accept: 'application/json',
+        'x-apikey': apiKey ?? ''
+      }
+    });
+
+    return await instance.get('https://www.virustotal.com/api/v3/files/upload_url').then(res => {
+      return res.data.data;
+    });
```

Comment:
Just use `return this.instance.post('/files/upload_url')`. No need to create another instance, the one from the constructor is fine.

Reply:
Actually I think we could just move this call in `files` method and remove the `getURL` method.

Reply:
Hi, this one line of code solution is indeed more elegant. I have made the change, though the style checker requires it to be in multiple lines.